index.bs

```diff
@@ -286,65 +290,76 @@ When the `[[track]]` of a {{MediaStreamTrackProcessor}} |processor|
 {{TrackEnded|ends}}, the [=processorClose=] algorithm must be
 executed with |processor| as parameter.
 
-## MediaStreamTrackGenerator ## {#generator}
-A {{MediaStreamTrackGenerator}} allows the creation of a {{WritableStream}}
-that acts as a {{MediaStreamTrack}} source in the
+
+## VideoTrackSource ## {#video-track-source}
+A {{VideoTrackSource}} allows the creation of a video source for a
+{{MediaStreamTrack}} in the
 <a href="https://www.w3.org/TR/mediacapture-streams/#the-model-sources-sinks-constraints-and-settings">
-MediaStream model</a>. Since the model does not expose sources directly
-but through the tracks connected to it, a {{MediaStreamTrackGenerator}}
-is also a track connected to its {{WritableStream}} source. Further tracks
-connected to the same {{WritableStream}} can be created using the
-{{MediaStreamTrack/clone}} method. The {{WritableStream}} source is
-exposed as the {{MediaStreamTrackGenerator/writable}} field of
-{{MediaStreamTrackGenerator}}.
-
-Similarly to {{MediaStreamTrackProcessor}}, the {{WritableStream}} of
-an audio {{MediaStreamTrackGenerator}} accepts {{AudioData}} objects,
-and a video {{MediaStreamTrackGenerator}} accepts {{VideoFrame}} objects.
-When a {{VideoFrame}} or {{AudioData}} object is written to
-{{MediaStreamTrackGenerator/writable}},
+MediaStream model</a> in the dedicated worker environment. It has a two readonly
+attributes: a {{VideoTrackSource/writable}} {{WritableStream}} and a
+{{VideoTrackSource/track}} {{MediaStreamTrack}}.
+
+The {{VideoTrackSource}} is the [=underlying sink=] of its
+{{VideoTrackSource/writable}} attribute. The {{VideoTrackSource/track}} attribute
+is the output. Further tracks connected to the same {{VideoTrackSource}} can be
+created using the {{MediaStreamTrack/clone}} method on the
+{{VideoTrackSource/track}} attribute.
+
+The {{WritableStream}} accepts {{VideoFrame}} objects.
+When a {{VideoFrame}} is written to {{VideoTrackSource/writable}},
 the frame's `close()` method is automatically invoked, so that its internal
 resources are no longer accessible from JavaScript.
 
-### Interface definition ### {#generator-interface}
-<pre class="idl">
-[Exposed=Window,DedicatedWorker]
-interface MediaStreamTrackGenerator : MediaStreamTrack {
-    constructor(MediaStreamTrackGeneratorInit init);
-    attribute WritableStream writable;  // VideoFrame or AudioFrame
-};
+Note: There is consensus in the WG that a source capable of generating a MediaStreamTrack of kind "video" should exist.
+There is no consensus in the WG about whether or not a source capable of generating a MediaStreamTrack of kind "audio" should exist.
 
-dictionary MediaStreamTrackGeneratorInit {
-  required DOMString kind;
+
+### Interface definition ### {#video-source-interface}
+<pre class="idl">
+[Exposed=DedicatedWorker]
+interface VideoTrackSource {
+  constructor();
+  readonly attribute WritableStream writable;
```

Comment:
We have Promise<XYZ>, could we write it as WritableStream<VideoFrame>?

Reply:
Don't understand this comment. What are you suggesting?